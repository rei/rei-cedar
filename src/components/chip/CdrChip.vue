<script setup>
import { useCssModule, useSlots } from 'vue';

const slots = useSlots();
const baseClass = 'cdr-chip';
const style = useCssModule();
const hasIconLeft = slots['icon-left'];
const hasIconRight = slots['icon-right'];
</script>

<template>
  <button
    :class="style[baseClass]"
  >
    <span
      v-if="hasIconLeft"
      :class="style['cdr-chip__icon-left']"
    >
      <slot name="icon-left" />
    </span>
    <span
      v-if="hasIconRight"
      :class="style['cdr-chip__icon-right']"
    >
      <slot name="icon-right" />
    </span>
    <span :class="style['cdr-chip__content']">
      <slot />
    </span>
  </button>
</template>

<style lang="scss" module src="./styles/CdrChip.module.scss">
</style>
