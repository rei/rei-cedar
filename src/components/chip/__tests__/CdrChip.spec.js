import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrChip from '../CdrChip.vue';

describe('CdrChip', () => {
  it('default matches snapshot', () => {
    const wrapper = mount(CdrChip, {
      slots: {
        default: 'tortilla chip'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders icon slots', () => {
    const wrapper = mount(CdrChip, {
      slots: {
        default: 'tortilla chip',
        'icon-left': '🌮',
        'icon-right': '🍔'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
