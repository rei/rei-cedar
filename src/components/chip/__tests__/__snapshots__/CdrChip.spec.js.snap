// Vitest Snapshot v1

exports[`CdrChip > default matches snapshot 1`] = `
<button
  class="cdr-chip"
>
  <!--v-if-->
  <!--v-if-->
  <span
    class="cdr-chip__content"
  >
    
    tortilla chip
    
  </span>
</button>
`;

exports[`CdrChip > renders icon slots 1`] = `
<button
  class="cdr-chip"
>
  <span
    class="cdr-chip__icon-left"
  >
    
    🌮
    
  </span>
  <span
    class="cdr-chip__icon-right"
  >
    
    🍔
    
  </span>
  <span
    class="cdr-chip__content"
  >
    
    tortilla chip
    
  </span>
</button>
`;

exports[`CdrChip default matches snapshot 1`] = `
<button
  class="cdr-chip"
>
  <!--v-if-->
  <!--v-if-->
  <span
    class="cdr-chip__content"
  >
    
    tortilla chip
    
  </span>
</button>
`;

exports[`CdrChip renders icon slots 1`] = `
<button
  class="cdr-chip"
>
  <span
    class="cdr-chip__icon-left"
  >
    
    🌮
    
  </span>
  <span
    class="cdr-chip__icon-right"
  >
    
    🍔
    
  </span>
  <span
    class="cdr-chip__content"
  >
    
    tortilla chip
    
  </span>
</button>
`;
